### DISCUSSION TITLE
Getting the color of a corner pixel and using it to make a frame

### DISCUSSION THREAD

**Discussion by lkintact** [ImageMagick/ImageMagick#7666] [Help] [answered]
### ImageMagick version

7.1.1-38 Q16-HDRI x64 b0ab922:20240901

### Operating system, version and so on

Windows 10 Home

### Description

Hi you guys,

I want to modify a series of images by adding a frame around them, but the color of the frame must be that of the pixel in the top-right corner. While with the help of [Examples of ImageMagick Usage](https://www.imagemagick.org/Usage/thumbnails/#frame) I can figure out how to make a frame, I don't know how to get the color of a corner pixel. Is it possible in IM?

**Comment by snibgo** [comment, marked as answer]
```
magick toes.png -mattecolor %[pixel:p[0,0]] -frame 20x10 x.png
```
The result, x.png, is:
![x](https://github.com/user-attachments/assets/fa13400e-76ca-4daa-b253-d6b93ed1447d)

EDIT: Sorry, I see you wanted the top-right corner. So:
```
magick toes.png -mattecolor %[pixel:p{w-1,0}] -frame 20x10 x2.jpg
```
![x2](https://github.com/user-attachments/assets/bc72d9bb-122f-4117-942a-02761d39997b)

**Comment by lkintact** [comment]
@snibgo Great, thank you!

**Comment by snibgo** [comment]
We cross-posted. See my edit.
